### DISCUSSION TITLE
Make vanilla-extract work on build time

### DISCUSSION THREAD

**Discussion by targetlucked69** [nuxt/nuxt#17453] [Questions] [answered]
Using https://vanilla-extract.style in Nuxt via their vite plugin works smoothly. But when running `nuxt build`, I think nuxt's postcss and vanilla-extract's postcss configs clash?

Related issues:

https://github.com/seek-oss/vanilla-extract/issues/445
https://github.com/seek-oss/vanilla-extract/issues/446

Here's the part where it throws: `Unknown word`

https://github.com/seek-oss/vanilla-extract/blob/master/packages/vite-plugin/src/index.ts#L103

**Comment by danielroe** [comment, marked as answer]
~> https://github.com/seek-oss/vanilla-extract/pull/729

**Reply by wobsoriano** [reply]
Tried the fix locally and the build works! But running `node .output/server/index.mjs` gives you this:

```bash
[nuxt] [request error] hash is not a function
  at generateIdentifier (./.output/server/chunks/app/server.mjs:353:23)
  at createTheme (./.output/server/chunks/app/server.mjs:482:24)
  at ./.output/server/chunks/app/server.mjs:4152:28
  at ModuleJob.run (node:internal/modules/esm/module_job:197:25)
  at async Promise.all (index 0)
  at async ESMLoader.import (node:internal/modules/esm/loader:337:24)
  at async ./.output/server/chunks/handlers/renderer.mjs:11146:24
  at async ./.output/server/chunks/handlers/renderer.mjs:11202:64
  at async ./.output/server/node_modules/h3/dist/index.mjs:417:19
  at async Server.nodeHandler (./.output/server/node_modules/h3/dist/index.mjs:367:7)
^C%
```

**Reply by danielroe** [reply]
You need to transpile some of vanilla extract's dependencies: see https://stackblitz.com/edit/github-8tflpr

**Reply by wobsoriano** [reply]
Yep that definitely fixed it

**Reply by targetlucked69** [reply]
Nice! The update works now, only problem is caching I guess. I've added my style files to `watch` option and though the client detects the update, the styles are still cached. I have to restart server to see changes

**Reply by danielroe** [reply]
@walmartwarlord Worth raising an issue either in Nuxt or vanilla-extract - not sure where the issue is arising but it's worth looking into it. I wouldn't think you should need to explicitly watch your style files - vite should take care of that for you.
